src/mpid/ch4/src/ch4i_workq.h

```diff
@@ -13,5 +13,133 @@
+    pt2pt_elemt->pt2pt.addr = addr;
+    pt2pt_elemt->pt2pt.status = status;
+    pt2pt_elemt->pt2pt.request = request;
+    pt2pt_elemt->pt2pt.flag = flag;
+    pt2pt_elemt->pt2pt.message = message;
+
+    MPIDI_workq_enqueue(workq, pt2pt_elemt);
+}
+
+MPL_STATIC_INLINE_PREFIX void MPIDI_workq_rma_enqueue(MPIDI_workq_t * workq,
+                                                      MPIDI_workq_op_t op,
+                                                      const void *origin_addr,
+                                                      int origin_count,
+                                                      MPI_Datatype origin_datatype,
+                                                      void *result_addr,
+                                                      int result_count,
+                                                      MPI_Datatype result_datatype,
+                                                      int target_rank,
+                                                      MPI_Aint target_disp,
+                                                      int target_count,
+                                                      MPI_Datatype target_datatype,
+                                                      MPI_Op acc_op,
+                                                      MPIR_Group * group,
+                                                      int lock_type,
+                                                      int assert,
+                                                      MPIR_Win * win_ptr,
+                                                      MPIDI_av_entry_t * addr,
+                                                      OPA_int_t * processed)
+{
+    MPIDI_workq_elemt_t *rma_elemt = NULL;
+    rma_elemt = MPIDI_workq_elemt_create();
+    rma_elemt->op = op;
+    rma_elemt->processed = processed;
+    rma_elemt->rma.origin_addr = origin_addr;
+    rma_elemt->rma.origin_count = origin_count;
+    rma_elemt->rma.origin_datatype = origin_datatype;
+    rma_elemt->rma.result_addr = result_addr;
+    rma_elemt->rma.result_count = result_count;
+    rma_elemt->rma.result_datatype = result_datatype;
+    rma_elemt->rma.target_rank = target_rank;
+    rma_elemt->rma.target_disp = target_disp;
+    rma_elemt->rma.target_count = target_count;
+    rma_elemt->rma.target_datatype = target_datatype;
+    rma_elemt->rma.acc_op = acc_op;
+    rma_elemt->rma.group = group;
+    rma_elemt->rma.lock_type = lock_type;
+    rma_elemt->rma.assert = assert;
+    rma_elemt->rma.win_ptr = win_ptr;
+    rma_elemt->rma.addr = addr;
+
+    MPIDI_workq_enqueue(workq, rma_elemt);
+}
+
+MPL_STATIC_INLINE_PREFIX void MPIDI_workq_release_pt2pt_elemt(MPIDI_workq_elemt_t * workq_elemt)
+{
+    MPIR_Request *req;
+    req = MPL_container_of(workq_elemt, MPIR_Request, dev.ch4.command);
+    MPIR_Request_free(req);
+}
+
+MPL_STATIC_INLINE_PREFIX int MPIDI_workq_dispatch(MPIDI_workq_elemt_t * workq_elemt)
```

Comment:
Will this `MPIDI_workq_dispatch` be needing a macro define down below when WORKQ is disabled?

Reply:
I looked PR #3117, I think this is fine.